omnipaxos/src/ballot_leader_election.rs

```diff
@@ -103,15 +121,20 @@ impl BallotLeaderElection {
         let num_nodes = &peers.len() + 1;
         let quorum = Quorum::with(config.flexible_quorum, num_nodes);
         let initial_ballot = Ballot::with(config_id, 0, config.priority, pid);
+        let initial_leader = match initial_leader {
+            Some(b) if b == Ballot::default() => initial_ballot,
+            Some(b) => b,
+            None => initial_ballot,
+        };
         let mut ble = BallotLeaderElection {
             configuration_id: config_id,
             pid,
             peers,
             hb_round: 0,
             heartbeat_replies: Vec::with_capacity(num_nodes),
             current_ballot: initial_ballot,
-            happy: true, //TODO: can this be false if there is an initial leader?
-            leader: Some(initial_ballot), //TODO: is this ok
+            happy: true,
```

Comment:
I think this is similar to when we set connectivity to the max initially. It's safe to set to true (if a leader doesn't get majority somehow it will go unhappy) and allows for an initial leader to be chosen in the first round.